File: 2575-find-the-divisibility-array-of-a-string/2575-find-the-divisibility-array-of-a-string.cpp
class Solution {
public:
    vector<int> divisibilityArray(string word, int m) {
        
        string str = "";
        vector<int>res(word.size(),0);
        long long num = 0;
        for(int i = 0; i<word.size(); i++){
            // str += word[i];
            num = (num * 10 + (word[i] - '0') ) % m;
            if(num == 0){
                res[i] = 1;
            }
        }
        
        return res;
    }
};

File: 2575-find-the-divisibility-array-of-a-string/2575-find-the-divisibility-array-of-a-string.ts
function divisibilityArray(word: string, m: number): number[] {
    
    let res : number[] = new Array(word.length).fill(0);
    let num : number = 0;
    
    for(let i = 0; i<word.length; i++){
        num = (num * 10 + parseInt(word[i])) %  m;
        if(num === 0){
            res[i] = 1;
        }
    }
    
    return res;
};